### PR TITLE
kola/tests: rewrite linux.nfs.* to use CLC

### PR DESCRIPTION
**Pull request by arithx**
`linux.nfs.v3` & `linux.nfs.v4` use cloudconfig. This ports them
to Container Linux Configs.

### CODE REVIEW

**Comment by ajeddeloh** [comment]
Did you figure out why it was flaking on `aws` (not that it wasn't flaky before) or is that still an open question?

**Comment by arithx** [comment]
The issue I was seeing on AWS was `systemd-hostnamed` failing correctly because it didn't have the permissions to read the `/etc/hostname` file (the CLC wasn't specifying the mode so it was set to [mode 0](https://github.com/coreos/bugs/issues/2208))

File: kola/tests/misc/nfs.go [hunk 1]
```diff
@@ -19,54 +19,28 @@ import (
...
+	nfsserverconf = conf.ContainerLinuxConfig(`storage:
+  files:
+    - filesystem: "root"
+      path: "/etc/hostname"
+      contents:
+        inline: "nfs1"
+      mode: 0644
+    - filesystem: "root"
+      path: "/etc/exports"
+      contents:
```
Comment (bgilbert) [inline kola/tests/misc/nfs.go]: Also mode 0644 for future-proofing.

Comment (arithx) [reply kola/tests/misc/nfs.go]: Pushed a change to add this. Good to merge on my end.

**Pull request merged**